flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobEdge.java

```diff
@@ -27,7 +28,7 @@
  * intermediate result partition to a job vertex. An edge is parametrized with its {@link
  * DistributionPattern}.
  */
-public class JobEdge implements java.io.Serializable {
+public class JobEdge implements LogicalEdge, java.io.Serializable {
```

Comment:
Let's avoid to change the inheritance of JobEdge to avoid potential compatibility issues.
We can introduce a `DefaultLogicalEdge` which is created from `JobEdge`.

Reply:
Thank you for proposing this idea. Resolved.